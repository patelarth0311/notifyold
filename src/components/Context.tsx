import { Dispatch, SetStateAction, createContext } from "react";
import { AppStatus } from "../types/types";
export const MyContext = createContext<{ appStatus: AppStatus, setAppStatus: (e: AppStatus) => void }>({
    appStatus: { response: "", status: "", userId: null },
    setAppStatus: (e: AppStatus) => { }
});