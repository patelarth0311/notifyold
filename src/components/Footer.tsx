import github from "../svgs/github.svg";
import linkedin from "../svgs/linkedin.svg";

export function Footer() {
  return (
    <div className="flex flex-col  bg-indigo-600 justify-center gap-y-3 items-center h-[140px]">
      <h1 className="text-xl font-ver text-center text-white">Notify</h1>
      <div className="flex  gap-x-4 ">
        <a
          target="_blank"
          rel="noopener noreferrer"
          href="https://github.com/patelarth0311"
        >
          <img width={25} src={github}></img>
        </a>
        <a
          target="_blank"
          rel="noopener noreferrer"
          href="https://www.linkedin.com/in/patelrarth/"
        >
          <img width={25} src={linkedin}></img>
        </a>
      </div>
    </div>
  );
}
