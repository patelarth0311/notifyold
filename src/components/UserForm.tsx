import { User, UserFormFields } from "../types/types";
import { PinInput, PinInputField } from "@chakra-ui/react";
import { HStack } from "@chakra-ui/react";
import { useState } from "react";
import { confirmUser } from "../API/Requests";

export function UserForm(
  props: UserFormFields & { showErrorMessage: boolean }
) {
  const [interacted, setInteracted] = useState(false);

  return (
    <div>
      <div className="flex font-ver justify-center items-center flex-col gap-y-20 w-full ">
        <div className="flex justify-center items-center flex-col gap-y-16 ">
          <div className="relative">
            <p className="absolute top-[-30px] text-[14px] ">Email</p>

            <input
              name="email"
              value={props.email}
              onChange={props.setForm}
              title="Email"
              type="email"
              className="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
            ></input>
          </div>

          <div className="relative">
            <p
              className={
                "absolute top-[-30px] text-[14px] " +
                (props.showErrorMessage ? "text-red-600" : "")
              }
            >
              Password
            </p>
            <input
              name="password"
              value={props.password}
              onFocus={(e) => {
                setInteracted(true);
              }}
              onChange={props.setForm}
              title="Password"
              type="password"
              className="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
            ></input>
          </div>
          {props.hasConfirmPassword && (
            <div className="relative flex flex-col gap-y-3">
              <p
                className={
                  "absolute top-[-30px] text-[14px] " +
                  (props.showErrorMessage ? "text-red-600" : "")
                }
              >
                Confirm Password
              </p>
              <input
                disabled={props.password.length === 0}
                name="confirmPassword"
                value={props.confirmPassword}
                onChange={props.setForm}
                title="Password"
                type="password"
                className={
                  "w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border " +
                  (props.password.length === 0 ? "bg-slate-100" : "")
                }
              ></input>
              {(props.showErrorMessage || interacted) && (
                <div className="text-[12px]  w-full justify-start  flex flex-col items-start h-full">
                  <p className="text-[13px]">Password must contain:</p>

                  <p>At least one lowercase and uppercase letter</p>
                  <p>At least one special character</p>
                  <p>At least one number</p>
                  <p>Be at least 8 characters</p>
                </div>
              )}
            </div>
          )}
        </div>

        <div className="flex flex-col gap-y-5">
          <button
            onClick={props.buttonAction}
            className="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
          >
            {props.buttonText}
          </button>

          <div className="flex w-[353px]  items-center justify-end gap-x-3 ">
            <p>{props.messageText}</p>
            <button
              onClick={props.setShow}
              className="text-base text-indigo-500 font-ver"
            >
              {props.optionText}
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

export function ConfirmationPin(props: { user: User; register: () => void }) {
  return (
    <div className="font-ver relative">
      <p className="absolute top-[-30px] text-[14px]">Enter code</p>
      <HStack>
        <PinInput
          onComplete={(e) => {
            confirmUser(props.user.username, e).then((res) => {
              console.log(res);
              if (res && res.status === 200) {
                props.register();
              }
            });
          }}
          type="number"
          placeholder=""
        >
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px] " />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
        </PinInput>
      </HStack>
    </div>
  );
}
