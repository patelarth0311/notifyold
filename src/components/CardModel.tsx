import { Entry, Model, Note, AppStatus } from "../types/types";
import { useEffect, useState } from "react";
import { useRef } from "react";
import close from "../svgs/close.svg";
import edit from "../svgs/edit.svg";
import plus from "../svgs/plus.svg";
import {
  deleteNote,
  postEntry,
  deleteEntry,
  changeEntry,
} from "../API/Requests";
import cal from "../svgs/calendar.svg";
import priority from "../svgs/priority.svg";
import { useContext } from "react";
import { MyContext } from "./Context";
import "../App.css";
import trash from "../svgs/trash.svg";
import { useToast } from "@chakra-ui/react";
import check from "../svgs/check.svg";
import minus from "../svgs/minus.svg";
import { makeToast } from "./Toast";
import flag from "../svgs/flag.svg"

export function CardModel(props: Model & { note: Note }) {
  const [editMode, setEditMode] = useState(false);
  const toast = useToast();
  const { appStatus, setAppStatus } = useContext(MyContext);
  const [createEntry, setCreateEntry] = useState(false);

  const [editEntry, setEditEntry] = useState({
    content: "",
    flag: false,
    time: "",
    priority: 0,
  });

  const [entry, setEntry] = useState({
    content: "",
    flag: false,
    time: "",
    priority: 0,
  });

  const handleChange = (e: any) => {
    if (e.target) {
      const { name, value, type, checked } = e.target;
    
      setEntry((prev) => {
        return {
          ...prev,
          [name]: type === "checkbox" ? checked : value,
        };
      });
    }
  };

  return (
    <div className="flex gap-y-4 flex-col p-10 ">
      <div className="flex gap-x-3 justify-end ">
        <button
          onClick={() => {
            deleteNote(
              props.note.noteId,
              appStatus.userId ? appStatus.userId : ""
            ).then((res) => {
              if (res?.status === 200) {
                setAppStatus({
                  ...appStatus,
                  status: res.data.status,
                  response: res.data,
                });
                props.setOpen(false);
                makeToast(  <img className="absolute left-10" src={check}></img>,props.note.name,"border-[#6FCF97]",toast, "deleted!")
              }
            });
          }}
          className="shadow-2xl rounded-full bg-indigo-500 w-8 h-8  flex justify-center items-center "
        >
          <img width={15} src={trash}></img>
        </button>

        {props.note.content.length > 0 && (
          <button
            onClick={() => {
              setEditMode((prev) => !prev);
              setCreateEntry(false);
            }}
            className={
              "shadow-2xl rounded-full bg-indigo-500 w-8 h-8 flex justify-center items-center " +
              (editMode ? " animate-bounce " : "")
            }
          >
            <img width={15} src={edit}></img>
          </button>
        )}

        <button
          onClick={() => props.setOpen(false)}
          className="shadow-2xl rounded-full bg-indigo-500 w-8 h-8  flex justify-center items-center "
        >
          <img width={15} src={close}></img>
        </button>
      </div>

      <div
        className={
          "mobile:w-[355px]  relative shadow-custom mobile:h-[410px]   w-[493.5px] h-[517.5px]   aspect-square rounded-[21.67px] bg-white "
        }
      >
        <div className="flex justify-start p-[15px] flex-col h-full b">
          <div className="justify-start flex  h-full gap-y-2 flex-col ">
            <p className="text-[17px] text-left font-bold text-black sticky">
              {props.note.name}
            </p>
            <div className="overflow-y-auto h-full relative ">
              <div className="flex justify-start items-start flex-col overflow-hidden w-full gap-y-3 overflow-y-auto p-1">
                {props.note.content.map((item, index) => (
                  <EditEntryField
                    appStatus={appStatus}
                    setAppStatus={setAppStatus}
                    noteId={props.note.noteId}
                    key={item.entryId}
                    item={item}
                    editMode={editMode}
                    editEntry={editEntry}
                    setEditEntry={(e: Entry) => {
                      setEditEntry(e);
                    }}
                  ></EditEntryField>
                ))}
                {createEntry && (
                  <CreateEntryField
                    entry={entry}
                    handleChange={handleChange}
                  ></CreateEntryField>
                )}
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className=" flex justify-between items-center w-full">
        <button
          onClick={() => {
            setEditMode(false);
            setCreateEntry((prev) => !prev);
          }}
          className={
            "  shadow-2xl rounded-full bg-indigo-500 w-8 h-8 flex justify-center items-center "
          }
        >
          <img width={15} src={plus}></img>
        </button>

        {(createEntry || editMode) && (
          <div className=" flex gap-x-1">
            {(props.note.content.length > 0 || createEntry) && (
              <button
                onClick={() => {
                  setEditMode(false);
                  setCreateEntry(false);
                  setEntry({ content: "", time: "", priority: 0, flag: false });
                }}
                className={
                  "bg-indigo-500 w-[90px] h-[30px] rounded-md text-white "
                }
              >
                Cancel
              </button>
            )}

            {createEntry && (
              <button
                onClick={() => {
                  try {
                    postEntry(
                      entry,
                      props.note.noteId,
                      appStatus.userId ? appStatus.userId : ""
                    ).then((res) => {
                      if (res?.status === 200) {
                        setAppStatus({
                          ...appStatus,
                          status: res.data.status,
                          response: res.data,
                        });
                        setEditMode(false);
                        setCreateEntry(false);
                        setEntry({
                          content: "",
                          time: "",
                          priority: 0,
                          flag: false,
                        });
                      }
                    });
                  } catch (err) {}
                }}
                className={
                  "bg-indigo-500 w-[90px] h-[30px]  rounded-md text-white "
                }
              >
                Submit
              </button>
            )}
          </div>
        )}
      </div>
    </div>
  );
}
function CreateEntryField(props: {
  entry: Entry;
  handleChange: (e: any) => void;
}) {
  return (
    <div className="flex flex-col gap-y-2 w-full">
      <input
        type="text"
        name="content"
        value={props.entry.content}
        onChange={props.handleChange}
        className="text-[15px] border p-1   relative rounded-md  "
        placeholder="Add info"
      ></input>
      <div className="relative flex w-full gap-x-10 ">
        <div className="flex gap-x-2  w-full">
          <img src={cal} width={20}></img>
          <input
            name="time"
            value={props.entry.time}
            onChange={props.handleChange}
            type="date"
            className=" border  w-full rounded-lg focus:outline-indigo-500 pl-1 p-1"
            placeholder="Select date"
          />
        </div>
        <div className="flex gap-x-2">
          <img width={20} src={priority}></img>

          <select
          defaultValue={props.entry.priority}
          
            onChange={props.handleChange}

            name="priority"
            placeholder="Select Priority"
          >
            <option value="0">None</option>
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
          </select>
        </div>

        <label className="flex gap-x-2">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            fill={props.entry.flag ? "#84cc16" : "#d1d5db"}
            className="w-14"
          >
            <path
              fillRule="evenodd"
              d="M3 2.25a.75.75 0 01.75.75v.54l1.838-.46a9.75 9.75 0 016.725.738l.108.054a8.25 8.25 0 005.58.652l3.109-.732a.75.75 0 01.917.81 47.784 47.784 0 00.005 10.337.75.75 0 01-.574.812l-3.114.733a9.75 9.75 0 01-6.594-.77l-.108-.054a8.25 8.25 0 00-5.69-.625l-2.202.55V21a.75.75 0 01-1.5 0V3A.75.75 0 013 2.25z"
              clipRule="evenodd"
            />
          </svg>

          <input
            value={props.entry.flag.toString()}
            type="checkbox"
            name="flag"
            onClick={props.handleChange}
            className="w-full h-full peer appearance-none rounded-md"
          />
        </label>
      </div>
    </div>
  );
}

function EditEntryField(props: {
  item: Entry;
  editMode: boolean;
  editEntry: Entry;
  setEditEntry: (e: Entry) => void;
  noteId: string;
  setAppStatus: (e: AppStatus) => void;
  appStatus: AppStatus;
}) {
  const [selectedField, setSelectedField] = useState(false);
  const inputRef = useRef<HTMLInputElement>(null);
  const [editEntry, setEditEntry] = useState({
    content: props.item.content,
    flag: props.item.flag,
    time: props.item.time,
    priority: props.item.priority,
  });

  useEffect(() => {

    const handleInput  = (e: Event) => {
      if (inputRef.current) {
        if (!inputRef.current.contains(e.target as Node)) {
          setSelectedField(false);
          if (!inputRef.current.contains(e.target as Node)) {
            setSelectedField(false);
          }
        }
      }
    }

    document.addEventListener("click", (e) => {
        handleInput(e)
    });

    return () => document.removeEventListener("click", (e) => {
        handleInput(e)
    });
  }, []);

  useEffect(() => {
    if (selectedField === false) {
      setEditEntry(props.item);
    }
  }, [selectedField]);

  return (
    <div className="flex flex-row gap-x-2 justify-between w-full ">
      <div className="w-full">
      <input
        ref={inputRef}
        value={editEntry.content}
        onClick={() => {
          setSelectedField((prev) => !prev);
        }}
        onKeyDown={(e) => {
          if (props.item.entryId && e.key === "Enter") {
            changeEntry(
              editEntry,
              props.noteId,
              props.appStatus.userId ? props.appStatus.userId : ""
            ).then((res) => {
              if (res?.status === 200) {
                props.setAppStatus({
                  ...props.appStatus,
                  status: res.data.status,
                  response: res.data,
                });
              }
            });
          }
        }}
        onChange={(e) => {
          setEditEntry((prev) => {
            return { ...prev, content: e.target.value };
          });
        }}
        className={
          "text-[15px] h-auto bg-white focus:outline-none relative rounded-md  w-full p-1 " +
          (props.editMode ? "  border" : "")
        }
        disabled={!props.editMode}
      ></input>
         <div className="flex  gap-x-1 p-1 items-center justify-start ">
            <p className="text-[11px] font-medium text-black text-left">{props.item.time}</p>
            <p className="text-[11px] font-medium  text-left text-[#f59e0b]">{"!".repeat(props.item.priority)}</p>
            {props.item.flag && (
              <img width={12} src={flag}></img>
            )
            }
              </div>
      </div>


      {props.editEntry && selectedField && (
        <button
          onClick={() => {
            if (props.item.entryId) {
              deleteEntry(
                props.noteId,
                props.item.entryId,
                props.appStatus.userId ? props.appStatus.userId : ""
              ).then((res) => {
                if (res?.status === 200) {
                  props.setAppStatus({
                    ...props.appStatus,
                    status: res.data.status,
                    response: res.data,
                  });
                }
              });
            }
          }}
          className="w-7 h-7 justify-center items-center"
        >
          <img height={2} src={minus}></img>
        </button>
      )}
    </div>
  );
}
