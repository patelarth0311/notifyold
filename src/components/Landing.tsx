import React from "react";
import { UserFormFields, User} from "../types/types";
import { useState } from "react";
import { addUser, signinUser } from "../API/Requests";
import { ConfirmationPin, UserForm } from "./UserForm";
import { useContext } from "react";
import { MyContext } from "./Context";
import { useToast } from "@chakra-ui/react";
import error from "../svgs/error.svg"
import check from "../svgs/check.svg"
import {
    CognitoUser,
    CognitoAccessToken,
    CognitoIdToken,
    CognitoRefreshToken,
    CognitoUserSession,
} from "amazon-cognito-identity-js";
import { makeToast } from "./Toast";


export function Landing() {
    
    const toast = useToast()
    const [showErrorMessage, setShowErrorMessage] = useState(false)
    
   
    const [showLogin, setShowLogin] = useState(true);
    const [formData, setFormData] = useState({email: "", password: "", confirmPassword: ""})
    const [showPin, setShowPin] = useState(false)
    const [user, setUser] = useState({userId: "", username: "", status: false} )
    const cognitoReg = /^(?!\s+)(?!.*\s+$)(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[$^*.[\]{}()?"!@#%&/\\,><':;|_~`=+\- ])[A-Za-z0-9$^*.[\]{}()?"!@#%&/\\,><':;|_~`=+\- ]{8,256}$/
    const {setAppStatus, appStatus} = useContext(MyContext)
    const [showPosMessage, setShowPosMessage] = useState<boolean |  undefined>(undefined)

    var updateForm = (e : any) => {

        if (e.currentTarget) {
            const {name, value}  = e.currentTarget
            setFormData((prev) => {
                return {
                    ...prev, [name] : value
                }
            })
        }
       

    }

    return  <div 
    className="flex justify-center pb-10 pt-10 min-h-screen flex-col bg-neutral-50 items-center gap-y-20">
        <h1 className="text-6xl font-ver">Notify</h1>

        {showLogin ?
        <UserForm showErrorMessage={showErrorMessage}
         setForm={updateForm}  
         
         buttonAction={() => {

            signinUser((userId: string) => (setAppStatus({...appStatus, userId: userId})), formData.email, formData.password, toast, () => {setFormData({email: "", password: "", confirmPassword: ""})})
           
         }}
                   
                   
                          

setShow={
            () => {
                setFormData({email: "", password: "", confirmPassword: ""})
                setShowLogin(prev => !prev)
            }}
            email={formData.email} password={formData.password} hasConfirmPassword={false} buttonText={"Log in"} messageText={"Don't have an account?"} optionText={"Sign up"}></UserForm>
            : <>
            {showPin ? <ConfirmationPin register={() => {
                
                setShowLogin(true)
                setShowPin(false)
                {makeToast(   <img className="absolute left-10" width={20} src={check}></img>,"Registration complete!","border-[#6FCF97]",toast)}

            }} user={user}></ConfirmationPin> :
                        <UserForm setForm={updateForm}  
                        showErrorMessage={showErrorMessage}
                        buttonAction={() => {
                            if (formData.password.match(cognitoReg)
                            && formData.confirmPassword === formData.password) {
                                setShowErrorMessage(false)
                                addUser(formData.email, formData.password, toast , () => {setFormData({email: "", password: "", confirmPassword: ""})})
                            } else if (formData.confirmPassword !== formData.password) {
                                

                                {makeToast(   <img className="absolute left-10" width={20} src={error}></img>,"Passwords do not match","border-[#dc2626]",toast)}
                                setShowErrorMessage(true)
                            } else {
                           
                                {makeToast(   <img className="absolute left-10" width={20} src={error}></img>,"Password criteria not met","border-[#dc2626]",toast)}
                                setShowErrorMessage(true)
                            }
                        }} setShow={
                            () => {
                                setFormData({email: "", password: "", confirmPassword: ""})
                                setShowLogin(prev => !prev)
                                setShowErrorMessage(false)
                                
                            }}
                            email={formData.email} confirmPassword={formData.confirmPassword} password={formData.password} hasConfirmPassword={!showLogin} buttonText={"Sign up"} messageText={"Have an account?"} optionText={"Log in"}></UserForm>

            }
            
            </>
            
        
        }


        </div>
 
}


