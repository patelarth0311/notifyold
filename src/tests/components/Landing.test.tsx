import React from "react";
import { cleanup, getByText, render, fireEvent, act} from '@testing-library/react';
import { Landing } from "../../components/Landing";
import { UserForm } from "../../components/UserForm";
import { shallow } from 'enzyme';

test('render landing', () => {
    const landing = render(<Landing/>);

   
    expect(landing.getByRole('button',{name: 'registrationoption'})).toHaveTextContent(/Sign up/)
    act(() => {
        landing.getByRole('button',{name: 'registrationoption'}).click()
    })

   expect(landing.getByRole('button',{name: 'registrationoption'})).toHaveTextContent(/Log in/)


    cleanup()
})



it ('landing snapshot', async () => {
    var landing =  render(<Landing/>);
  

    expect(landing).toMatchSnapshot()

    act(() => {
    landing.getByRole('button',{name: 'registrationoption'}).click()
    })
    landing =  render(<Landing/>)

    expect(landing).toMatchSnapshot()
    cleanup()
})

