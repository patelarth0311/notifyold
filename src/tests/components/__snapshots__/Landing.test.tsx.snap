// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center pb-10 pt-10 min-h-screen flex-col bg-neutral-50 items-center gap-y-20"
      >
        <h1
          class="text-6xl font-ver"
        >
          Notify
        </h1>
        <div>
          <div
            class="flex font-ver justify-center items-center flex-col gap-y-20 w-full "
          >
            <div
              class="flex justify-center items-center flex-col gap-y-16 "
            >
              <div
                class="relative"
              >
                <p
                  class="absolute top-[-30px] text-[14px] "
                >
                  Email
                </p>
                <input
                  aria-label="email"
                  class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                  name="email"
                  role="textbox"
                  title="Email"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="relative"
              >
                <p
                  class="absolute top-[-30px] text-[14px] "
                >
                  Password
                </p>
                <input
                  aria-label="password"
                  class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                  name="password"
                  title="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-y-5"
            >
              <button
                aria-label="process-user-button"
                class="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
              >
                Log in
              </button>
              <div
                class="flex w-[353px]  items-center justify-end gap-x-3 "
              >
                <p>
                  Don't have an account?
                </p>
                <button
                  aria-label="registrationoption"
                  class="text-base text-indigo-500 font-ver"
                >
                  Sign up
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center pb-10 pt-10 min-h-screen flex-col bg-neutral-50 items-center gap-y-20"
    >
      <h1
        class="text-6xl font-ver"
      >
        Notify
      </h1>
      <div>
        <div
          class="flex font-ver justify-center items-center flex-col gap-y-20 w-full "
        >
          <div
            class="flex justify-center items-center flex-col gap-y-16 "
          >
            <div
              class="relative"
            >
              <p
                class="absolute top-[-30px] text-[14px] "
              >
                Email
              </p>
              <input
                aria-label="email"
                class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                name="email"
                role="textbox"
                title="Email"
                type="text"
                value=""
              />
            </div>
            <div
              class="relative"
            >
              <p
                class="absolute top-[-30px] text-[14px] "
              >
                Password
              </p>
              <input
                aria-label="password"
                class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                name="password"
                title="Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="flex flex-col gap-y-5"
          >
            <button
              aria-label="process-user-button"
              class="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
            >
              Log in
            </button>
            <div
              class="flex w-[353px]  items-center justify-end gap-x-3 "
            >
              <p>
                Don't have an account?
              </p>
              <button
                aria-label="registrationoption"
                class="text-base text-indigo-500 font-ver"
              >
                Sign up
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`landing snapshot 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center pb-10 pt-10 min-h-screen flex-col bg-neutral-50 items-center gap-y-20"
      >
        <h1
          class="text-6xl font-ver"
        >
          Notify
        </h1>
        <div>
          <div
            class="flex font-ver justify-center items-center flex-col gap-y-20 w-full "
          >
            <div
              class="flex justify-center items-center flex-col gap-y-16 "
            >
              <div
                class="relative"
              >
                <p
                  class="absolute top-[-30px] text-[14px] "
                >
                  Email
                </p>
                <input
                  aria-label="email"
                  class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                  name="email"
                  role="textbox"
                  title="Email"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="relative"
              >
                <p
                  class="absolute top-[-30px] text-[14px] "
                >
                  Password
                </p>
                <input
                  aria-label="password"
                  class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                  name="password"
                  title="Password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="relative flex flex-col gap-y-3"
              >
                <p
                  class="absolute top-[-30px] text-[14px] "
                >
                  Confirm Password
                </p>
                <input
                  aria-label="password-confirm"
                  class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border bg-slate-100"
                  disabled=""
                  name="confirmPassword"
                  title="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-y-5"
            >
              <button
                aria-label="process-user-button"
                class="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
              >
                Sign up
              </button>
              <div
                class="flex w-[353px]  items-center justify-end gap-x-3 "
              >
                <p>
                  Have an account?
                </p>
                <button
                  aria-label="registrationoption"
                  class="text-base text-indigo-500 font-ver"
                >
                  Log in
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="flex justify-center pb-10 pt-10 min-h-screen flex-col bg-neutral-50 items-center gap-y-20"
      >
        <h1
          class="text-6xl font-ver"
        >
          Notify
        </h1>
        <div>
          <div
            class="flex font-ver justify-center items-center flex-col gap-y-20 w-full "
          >
            <div
              class="flex justify-center items-center flex-col gap-y-16 "
            >
              <div
                class="relative"
              >
                <p
                  class="absolute top-[-30px] text-[14px] "
                >
                  Email
                </p>
                <input
                  aria-label="email"
                  class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                  name="email"
                  role="textbox"
                  title="Email"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="relative"
              >
                <p
                  class="absolute top-[-30px] text-[14px] "
                >
                  Password
                </p>
                <input
                  aria-label="password"
                  class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                  name="password"
                  title="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-y-5"
            >
              <button
                aria-label="process-user-button"
                class="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
              >
                Log in
              </button>
              <div
                class="flex w-[353px]  items-center justify-end gap-x-3 "
              >
                <p>
                  Don't have an account?
                </p>
                <button
                  aria-label="registrationoption"
                  class="text-base text-indigo-500 font-ver"
                >
                  Sign up
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center pb-10 pt-10 min-h-screen flex-col bg-neutral-50 items-center gap-y-20"
    >
      <h1
        class="text-6xl font-ver"
      >
        Notify
      </h1>
      <div>
        <div
          class="flex font-ver justify-center items-center flex-col gap-y-20 w-full "
        >
          <div
            class="flex justify-center items-center flex-col gap-y-16 "
          >
            <div
              class="relative"
            >
              <p
                class="absolute top-[-30px] text-[14px] "
              >
                Email
              </p>
              <input
                aria-label="email"
                class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                name="email"
                role="textbox"
                title="Email"
                type="text"
                value=""
              />
            </div>
            <div
              class="relative"
            >
              <p
                class="absolute top-[-30px] text-[14px] "
              >
                Password
              </p>
              <input
                aria-label="password"
                class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                name="password"
                title="Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="flex flex-col gap-y-5"
          >
            <button
              aria-label="process-user-button"
              class="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
            >
              Log in
            </button>
            <div
              class="flex w-[353px]  items-center justify-end gap-x-3 "
            >
              <p>
                Don't have an account?
              </p>
              <button
                aria-label="registrationoption"
                class="text-base text-indigo-500 font-ver"
              >
                Sign up
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
